### ISSUE TITLE
Multi user support

### ISSUE THREAD

**Issue by pik4ez** [frenata/gomua#2]
It's now the beginning of gomua development. So, it's time to make core architecture decisions.

It'd be great to have a good native multi user support. I think modern MUA must have an ability to easily operate multiple mailboxes.

Have a look at mutt's implementation and never do it their way.

**Comment by frenata** [comment]
Thanks for the interest in the project. :) Yeah this would definitely be a good feature to have--multiple mailboxes in general feels like a must if only for sorting/shifting mail around, and it's on my list of things to work on.

I'll keep multi-user in mind as the mailbox architecture expands. Is your basic idea to be able to have a MDA pulling in mail from multiple email accounts, and GoMua capable of reading them all?

**Comment by pik4ez** [comment]
There's a chance simple solution might strike. Different mailboxes, ability to read from all in one interface, just as you said.

I'm scared of monsters like thunderbird, 'cause they almost always heavy, slow and greedy to the memory. But they have client side filtering, which is useful for me.

So, asynchronous reading from physically different mailboxes plus basic client side filtering should work for me. Filtering is optional anyway, and maybe it shouldn't be considered for some first releases.

I didn't say about interface, because it can be either one common list or switching between mailboxes with separate messages lists. It's not so important for me. Important is to have native multi boxes support without any hooks and without need to start a few apps in different terminals. Just set up and fly.

Thank you that you stay in touch. Good luck with project.

**Comment by frenata** [comment]
> Important is to have native multi boxes support without any hooks and without need to start a few apps in different terminals. Just set up and fly.

Yep, that's what I'd go for, I think. Probably first pass will just be adding some config lines. For now my priority is to continue to expand the test coverage and generally get the codebase in a better place than it's very clear 48hour hack legacy. :)

Of course you're more than welcome to contribute, if you're interested.
